pkg/kv/kvserver/batcheval/cmd_export_test.go

```diff
@@ -48,22 +49,29 @@ import (
 
 func TestExportCmd(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 	dir, dirCleanupFn := testutils.TempDir(t)
 	defer dirCleanupFn()
-	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{ServerArgs: base.TestServerArgs{ExternalIODir: dir}})
-	defer tc.Stopper().Stop(ctx)
-	kvDB := tc.Server(0).DB()
+	srv, db, kvDB := serverutils.StartServer(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109429),
+		ExternalIODir:     dir,
+	})
+	defer srv.Stopper().Stop(ctx)
+	ts := srv.ApplicationLayer()
+
+	sql.SecondaryTenantSplitAtEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
 
 	export := func(
 		t *testing.T, start hlc.Timestamp, mvccFilter kvpb.MVCCFilter, maxResponseSSTBytes int64,
 	) (kvpb.Response, *kvpb.Error) {
+		key := ts.Codec().TablePrefix(bootstrap.TestingUserDescID(0))
 		req := &kvpb.ExportRequest{
-			RequestHeader:  kvpb.RequestHeader{Key: bootstrap.TestingUserTableDataMin(), EndKey: keys.MaxKey},
+			RequestHeader:  kvpb.RequestHeader{Key: key, EndKey: keys.MaxKey},
```

Comment:
I don't think this is correct: if we're moving the start key into a tenant span obtained from ts.Codec, we need to move the end key as well

Reply:
How?

Reply:
maybe codec.TenantSpan().EndKey